File: src/com/suushiemaniac/cubing/alglib/lang/InvalidNotationErrorListener.java
package com.suushiemaniac.cubing.alglib.lang;

import com.suushiemaniac.cubing.alglib.exception.InvalidNotationException;
import org.antlr.v4.runtime.BaseErrorListener;
import org.antlr.v4.runtime.RecognitionException;
import org.antlr.v4.runtime.Recognizer;

public class InvalidNotationErrorListener extends BaseErrorListener {
    private String notation;

    public InvalidNotationErrorListener(String notation) {
        this.notation = notation;
    }

    @Override
    public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String msg, RecognitionException e) {
        throw new InvalidNotationException(this.notation, msg);
    }
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/lang/InvalidNotationErrorListener.kt
package com.suushiemaniac.cubing.alglib.lang

import com.suushiemaniac.cubing.alglib.exception.InvalidNotationException
import org.antlr.v4.runtime.BaseErrorListener
import org.antlr.v4.runtime.RecognitionException
import org.antlr.v4.runtime.Recognizer

class InvalidNotationErrorListener(private val notation: String) : BaseErrorListener() {
    override fun syntaxError(recognizer: Recognizer<*, *>?, offendingSymbol: Any?, line: Int, charPositionInLine: Int, msg: String?, e: RecognitionException?) {
        throw InvalidNotationException(this.notation, msg!!)
    }
}
